Lib/contextlib.py

```diff
@@ -54,8 +54,9 @@ def inner(*args, **kwds):
         return inner
 
 
-class _GeneratorContextManager(ContextDecorator, AbstractContextManager):
-    """Helper for @contextmanager decorator."""
+class _GeneratorContextManagerBase(ContextDecorator):
```

Comment:
ContextDecorator is designed to work with `__call__` rather than `__await__`, so having it also applied to `asynccontextmanager` doesn't look right to me.

So I'd suggest moving the ContextDecorator inheritance and the `_recreate_cm` method down into `_GeneratorContextManager` and only keeping the `__init__` method common between the two.